test/stdlib/Observation/Observable.swift

```diff
@@ -218,6 +218,40 @@ class GenericClassParent<T> {
   }
 }
 
+protocol MemberProtocol {}
+struct MemberType: MemberProtocol {}
+
+@Observable
+class HasUniversalMember {
+  var member: any MemberProtocol = MemberType()
+}
+
+@Observable
+class HasExistentialMember {
+  var member: some MemberProtocol = MemberType()
+}
+
+protocol HasAssociatedType {
+  associatedtype Associated
+  static var associatedMember: Associated { get }
+}
+
+@Observable
+class HasHiddenExistential: HasAssociatedType {
+  var member: Associated = HasHiddenExistential.associatedMember
+  static var associatedMember: some MemberProtocol { MemberType() }
+}
+
+/*
+// FIXME: This case is not handled yet. Properties that have no type annotation
```

Comment:
Question for reviewers: I don't know how best to handle this case.

The simplest thing to do is just disable the optimization when there's no explicit type, but that's an annoying performance hazard when it's idiomatic to use an inferred type for most default-initialized properties.

It's very simple to fix the rare case where this fails (just add a type annotation) but that's still leaving in a source break.